src/inference_result.cc

```diff
@@ -184,32 +185,44 @@ void InferenceResult::show_marginal_snippet(std::ostream &output) const {
   output << "   ..." << std::endl;
 }
 
-void InferenceResult::show_marginal_histogram(std::ostream &output) const {
+void InferenceResult::show_marginal_histogram(std::ostream &output,
+                                              const size_t &bins) const {
   // show a histogram of inference results
   output << "INFERENCE CALIBRATION (QUERY BINS):" << std::endl;
-  std::vector<num_variables_t> abc;
-  for (int i = 0; i <= 10; ++i) {
-    abc.push_back(0);
-  }
+  std::vector<num_variables_t> abc(bins + 1, 0);
+
   num_variables_t bad = 0;
   for (variable_id_t j = 0; j < nvars; ++j) {
     const Variable &variable = fg.variables[j];
     if (!opts.should_sample_evidence && variable.is_evid) {
       continue;
     }
-    int bin =
-        (int)((double)agg_means[variable.id] / agg_nsamples[variable.id] * 10);
-    if (bin >= 0 && bin <= 10) {
+    size_t bin = (size_t)((double)agg_means[variable.id] /
+                          agg_nsamples[variable.id] * bins);
+    if (bin >= 0 && bin <= bins) {
       ++abc[bin];
     } else {
       ++bad;
     }
   }
-  abc[9] += abc[10];
-  for (int i = 0; i < 10; ++i) {
-    output << "PROB BIN 0." << i << "~0." << (i + 1) << "  -->  # " << abc[i]
-           << std::endl;
+  abc[bins - 1] += abc[bins];
+
+  // save ostream settings
+  const std::ios::fmtflags flags(output.flags());
+  const size_t prec = output.precision();
+
+  // set format for numbers
+  output << std::fixed;  // specify number of decimals (rather than sig figs)
+  output << std::setprecision(1);  // 1 digit after decimal point
```

Comment:
Shouldn't we take the ceiling of log10 or so of `bins` instead of fixing it to 1?

Seems like the lines can be combined?
```
const streamsize prec = output.precision(newprec);
```
http://www.cplusplus.com/reference/ios/ios_base/precision/